// cobrowse-agent-ui/dist/Device.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Device;

var _react = _interopRequireDefault(require("react"));

var _i18n = _interopRequireDefault(require("./i18n"));

var _PlatformIcon = _interopRequireDefault(require("./PlatformIcon.js"));

var _deviceType = _interopRequireDefault(require("./deviceType.js"));

require("./Device.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Device(props) {
  function renderLastSeen() {
    if (props.device.online) {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "last-seen"
      }, _i18n.default.t('Online'));
    } else {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "last-seen"
      }, _i18n.default.t('Last seen {{date, dateRelative}}', {
        date: props.device.last_active
      }));
    }
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    style: props.style,
    className: "Device ".concat(props.device.online ? 'online' : '', " ").concat(props.device.connectable ? 'connectable' : '')
  }, /*#__PURE__*/_react.default.createElement(_PlatformIcon.default, {
    platform: props.device.device.platform,
    className: "platform-icon"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "details"
  }, (0, _deviceType.default)(props.device.device), renderLastSeen()), props.children);
}
