// cobrowse-agent-ui/dist/Session.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Session;

var _react = _interopRequireDefault(require("react"));

var _deviceType = _interopRequireDefault(require("./deviceType.js"));

var _Stopwatch = _interopRequireDefault(require("./Stopwatch.js"));

var _i18n = _interopRequireDefault(require("./i18n"));

var _reactI18next = require("react-i18next");

require("./Session.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Session(props) {
  function openRecording() {
    if (props.openRecording) props.openRecording(props.session);
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    style: props.style,
    onClick: props.onClick,
    className: "Session ".concat(props.className || '')
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "details"
  }, /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_reactI18next.Trans, {
    i18n: _i18n.default
  }, /*#__PURE__*/_react.default.createElement("span", {
    className: "device-prefix"
  }, "Connected to "), {
    deviceType: (0, _deviceType.default)(props.session.device)
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "subdetails"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "activated"
  }, _i18n.default.t('{{date, dateRelative}}', {
    date: props.session.activated
  })), props.session.state === 'ended' && props.session.recorded ? /*#__PURE__*/_react.default.createElement("div", {
    className: "recorded",
    onClick: openRecording
  }, _i18n.default.t('Recorded')) : null)), props.session.state === 'ended' ? /*#__PURE__*/_react.default.createElement(_Stopwatch.default, {
    className: "duration",
    start: props.session.activated,
    end: props.session.ended || new Date()
  }) : /*#__PURE__*/_react.default.createElement("div", {
    className: "active"
  }, _i18n.default.t('Active')), props.children);
}
